workflow-core/src/test/java/com/squareup/workflow/SinkTest.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runBlockingTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+@OptIn(
+    ExperimentalCoroutinesApi::class,
+    ExperimentalStdlibApi::class,
+    ExperimentalWorkflow::class
+)
+class SinkTest {
+
+  private val sink = RecordingSink()
+
+  @Test fun `collectToActionSink sends action`() {
+    runBlockingTest {
+      val flow = MutableStateFlow(1)
+      val collector = launch {
```

Comment:
Nit: collector is slightly overloaded here given the coroutine context.  Maybe sinkCollector? or collectorJob?